app/models/king.rb

```diff
@@ -12,9 +15,9 @@ def can_move_out_of_check?
       ((y_position - 1)..(y_position + 1)).each do |y|
         return true if valid_move?(x, y) &&
                        game.under_attack?(is_white, x, y) == false &&
-                       (game.attacking_piece.straight_obstruction_array(x, y) ? game.attacking_piece.straight_obstruction_array(x, y).include?([x_position, y_position]) == false : true)
-        false
+                       (game.attacking_piece.type != 'knight' ? game.attacking_piece.straight_move?(x, y) == false : true)
```

Comment:
I tried a different direction with the logic here. Because all pieces besides the knight move in a straight line, I could just rule out knights, then check to see if the attacking piece was in a straight line path to the square where the king might try to move. Then, even if the king currently is blocking the square where he might move from the attacking piece, this method will catch that square too.